cmd/template/app/flag.go

```diff
@@ -68,6 +71,12 @@ func (f *flag) Validate() error {
 	if !f.InCluster && f.Cluster == "" {
 		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagCluster)
 	}
+	// Once we fully migrate to org namespace, checking for
+	// the `organization` flag should be mandatory.
+	//
+	/*if !f.InCluster && f.Organization == "" {
+		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagOrganization)
+	}*/
```

Comment:
The problem is this migration will take many months. I think maybe we could have some kind of feature detection instead of flag validation.

Reply:
Sure, I believe we do not need this basic flag validation, I put it there just as an example of what we will eventually end up with. But for now we could treat this flag as a feature detection mechanism, since it presence does reflect of how the App CR is templated. Or we could add a boolean flag like `--capo` and then upon user providing it, demand and validate the `organization`. WDYT @rossf7 ?

Reply:
I would remove it for now but mention in the PR that we can't validate the flag.

> But for now we could treat this flag as a feature detection mechanism, since it presence does reflect of how the App CR is templated.

Yes lets use it as a feature flag for now as we only need it for CAPO.

> Or we could add a boolean flag like --capo and then upon user providing it, demand and validate the organization

We try to keep the app platform decoupled from the provider. I think our best option is to extend app-admission-controller and give the user a useful error if the cluster label is missing. WDYT?

Reply:
I actually very like the idea of extending the app-admission-controller

Reply:
Let's put it on hold then, I will submit a PR to app-admission-controller and link it here, so that we can then advance with this PR.

Reply:
WDYT @rossf7 ?

Reply:
Good call and there is also https://github.com/giantswarm/giantswarm/issues/20448 for that.

We can then drop the version label from the `default-apps-openstack` chart 🔥

Reply:
Ok, we got it: https://github.com/giantswarm/app/pull/206 and https://github.com/giantswarm/app-admission-controller/pull/222

Reply:
@ljakimczuk We're going to need some more changes to remove usage of the cluster namespace.

There is also the flag `--defaulting-enabled` when its false it renders more fields.

```
k gs template app \
  --catalog=giantswarm \
  --cluster=eggs2 \
  --name=nginx-ingress-controller-app \
  --namespace=kube-system \
  --version=1.17.0 \
  --defaulting-enabled=false \
  --organization=test
---
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 1.0.0
  name: nginx-ingress-controller-app
  namespace: org-test
spec:
  catalog: giantswarm
  config:
    configMap:
      name: eggs2-cluster-values
      namespace: eggs2
    secret:
      name: ""
      namespace: ""
  kubeConfig:
    context:
      name: eggs2-kubeconfig
    inCluster: false
    secret:
      name: eggs2-kubeconfig
      namespace: eggs2
  name: nginx-ingress-controller-app
  namespace: kube-system
  version: 1.17.0
```

The cluster values CM and the kubeconfig secret need to be in the org namespace. We should also check the defaulting logic in app-admission-controller.

Can you also check the output if the user sets `--user-configmap` or `--user-secret`?

Reply:
@rossf7 `--user-configmap` example is already given in the PR description, I will also add there another example with `--user-secret`, but indeed I missed the `defaulting-enabled` and  there seems to be something wrong with it

Reply:
> --user-configmap example is already given in the PR description

@ljakimczuk Ah my bad I missed that 👍

Reply:
@rossf7 my bad, there is still one issue there

Reply:
@rossf7 should be good now, WDYT?

Reply:
In the meantime I will check the defaulting in app admission controller